File: WebServices/app/src/main/java/com/catherine/webservices/interfaces/MainInterface.java
package com.catherine.webservices.interfaces;

import android.os.Bundle;
import android.view.View;

import com.catherine.webservices.network.NetworkHealthListener;

/**
 * Created by Catherine on 2017/11/29.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */

public interface MainInterface {
    void getPermissions(String[] permissions, OnRequestPermissionsListener listener);

    void switchTab(int position);

    void callFragment(int id);

    void callFragment(int id, Bundle bundle);

    void callFragmentDialog(int id);

    void callFragmentDialog(int id, Bundle bundle);

    void hideKeyboard();

    void openSlideMenu();

    void backToPreviousPage();

    void clearAllFragments();

    void setBackKeyListener(BackKeyListener listener);

    void removeBackKeyListener();

    void addBottomLayout(int id);

    View getBottomLayout();

    void restoreBottomLayout();

    void listenToNetworkState(NetworkHealthListener listener);

    void stopListeningToNetworkState(NetworkHealthListener listener);
}

File: WebServices/app/src/main/java/com/catherine/webservices/interfaces/MainInterface.kt
package com.catherine.webservices.interfaces

import android.os.Bundle
import android.view.View


/**
 * Created by Catherine on 2017/8/28.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */
interface MainInterface {
    fun getPermissions(permissions: Array<String>, listener: OnRequestPermissionsListener)
    fun switchTab(position: Int)
    fun callFragment(id: Int)
    fun callFragment(id: Int, bundle: Bundle?)
    fun hideKeyboard()
    fun openSlideMenu()
    fun backToPreviousPage()
    fun clearAllFragments()
    fun setBackKeyListener(listener: BackKeyListener)
    fun removeBackKeyListener()
    fun addBottomLayout(id: Int)
    fun getBottomLayout(): View
    fun restoreBottomLayout()
}